artemis-commons/src/main/java/org/apache/activemq/artemis/utils/critical/CriticalAnalyzerImpl.java

```diff
@@ -48,11 +51,28 @@ public CriticalAnalyzerImpl() {
        *  issues and won't be able to shutdown the server or halt the VM
        */
       this.scheduledComponent = new ActiveMQScheduledComponent(null, null, checkTimeNanoSeconds, TimeUnit.NANOSECONDS, false) {
+
          @Override
          public void run() {
             logger.trace("Checking critical analyzer");
             check();
          }
+
+         @Override
+         protected ActiveMQThreadFactory getThreadFactory() {
+            return new ActiveMQThreadFactory("CriticalAnalyzer", "Critical-Analyzer-", true, getThisClassLoader());
```

Comment:
How come the thread needs to be a daemon thread now?

Reply:
The thought was that the critical analyzer should never prevent shutdown.  Catching the exception fixes this one case; a daemon thread is additional protection.

Reply:
@jbertram I was actually talking to @sjhiggs .. I think these thread should always been daemon. It was a mistake not to.

Reply:
I was mixed up on how daemon threads related to the JVM exiting. Sorry for the confusion. This looks good.